plugins/parquet/parquetembed.cpp

```diff
@@ -105,13 +105,69 @@ static bool toEclCompatibleFieldName(StringBuffer &name, StringBuffer &out)
             out.append(c);
     }
 
-    // Create a simple hash from the original name for uniqueness
-    unsigned hash = hashc((const unsigned char *)name.str(), name.length(), 0);
-    out.appendf("_%x", hash & 0xFFF); // limit to 3 hex digits
-
     return false;
 }
 
+/**
+ * @brief Retrieves the field names from an Arrow schema and stores them in a vector of StringBuffers.
+ *
+ * @param schema Shared pointer to the Arrow schema.
+ * @return Shared pointer to a flattened list of StringBuffers containing all field names.
+ */
+static std::shared_ptr<std::vector<StringBuffer>> getAllFieldNames(arrow::Schema &schema)
+{
+    auto originalNames = std::make_shared<std::vector<StringBuffer>>();
+
+    int numFields = schema.num_fields();
+    for (int i = 0; i < numFields; i++)
+    {
+        const std::string &fieldName = schema.field(i)->name();
+        originalNames->emplace_back(fieldName.length(), fieldName.c_str());
+    }
+
+    return originalNames;
+}
+
+/**
+ * @brief Generates a list of ParquetFieldInfo objects with unique ECL-compatible field names. Does not explore nested structures.
+ * Caller should get child schema and call getUniqueFieldNames as needed for nested structures.
+ * If a field name is not ECL-compatible, it is transformed and made unique by appending a numeric suffix if necessary.
+ *
+ * @param schema Shared pointer to the Arrow schema.
+ * @return ParquetFieldList A list of ParquetFieldInfo objects representing the fields with unique ECL-compatible names.
+ */
+static std::shared_ptr<ParquetFieldList> getUniqueFieldNames(arrow::Schema &schema)
+{
+    std::shared_ptr<std::vector<StringBuffer>> uniqueFieldNames = getAllFieldNames(schema);
```

Comment:
I don't think this needs to be a shared_ptr.  That only really needs to be used if there will be multiple references to the same object.